### ISSUE TITLE
"~username" in "file" not expanded

### ISSUE THREAD

**Issue by Lx** [RHInception/ansible-xml#13]
Given a call like this:

```
- xml: file=~user/data.xml ...
```

the following error occurs:

```
The target XML source does not exist: ~user/data.xml
```

even though built-in Ansible modules correctly expand `~user` to the correct thing (which may not necessarily be `/home/user`).

**Comment by Lx** [comment]
I see that this issue is addressed by pending pull request #10.

**Comment by cmprescott** [comment]
I wouldn't hold your breath for them to merge in my PR. This project seems to be dead. If you need a workaround you can do something like `{{ path | expanduser }}`. More details on http://docs.ansible.com/playbooks_filters.html

**Comment by Lx** [comment]
Thanks for this workaround—much tidier than mine, which involved `register`-ing a `shell` call to `echo ~user`.

I hope development picks up again, because there doesn't seem to be another robust solution at this time for reading and writing XML from Ansible tasks.

**Comment by Lx** [comment]
Caveat for anyone else following: templates are rendered locally, so `expanduser` operates on data from the **local** machine, not the remote machine.  To retrieve a home directory path for an arbitrary user on the remote machine, I'm back to doing this:

```yaml
- name: note absolute home directory location for XML manipulation
  shell: echo ~user
  changed_when: false
  register: user_home_dir
```

**Comment by tbielawa** [comment]
We're going to make an announcement about our livelihood very soon :-)

**Comment by tbielawa** [comment]
Closed when #10 was merged
